compiler/rustc_resolve/src/check_unused.rs

```diff
@@ -373,7 +373,14 @@ impl Resolver<'_, '_> {
                 }
             );
 
-            let fix_msg = if fixes.len() == 1 && fixes[0].0 == unused.item_span {
+            let fix_msg = if let [(sp, _)] = fixes.as_mut_slice()
+                && *sp == unused.item_span
+            {
+                // If this is the only import on the line, remove the whole line
+                if let Ok(next) = tcx.sess.source_map().span_to_next_source(*sp) {
+                    let whitespace = next.chars().take_while(char::is_ascii_whitespace).count();
+                    *sp = sp.with_hi(sp.hi() + rustc_span::BytePos(whitespace as u32));
```

Comment:
`whitespace` here is counted in `char`s, not bytes.

Reply:
Are there any other suggestions that remove full lines?
There probably are, in that case would be better to share this logic with them.

Reply:
Given that the iterator takes only `ascii_whitespace`, I think it makes sense to treat the char count as byte count? I'll look for similar suggestions